app.rb

```diff
@@ -0,0 +1,139 @@
+      @people.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+    @main_call.display_menu
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    person_type = gets.chomp.to_i
+    case person_type
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Invalid option'
+    end
+    @main_call.display_menu
+  end
+
+  def create_student
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Has parent permission? [Y/N]:'
+    parent_permission = gets.chomp.upcase == 'y'
+    @people.push(Student.new(age, name, parent_permission: parent_permission))
+    puts
+    puts "Person #{name} was created"
+  end
+
+  def create_teacher
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Specialization:'
+    specialization = gets.chomp
+    @people.push(Teacher.new(age, specialization, name))
+    puts
+    puts "Person #{name} was created"
+  end
+
+  def create_book
+    puts 'Enter the book title:'
+    title = gets.chomp
+    puts 'Enter the book author:'
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+    @main_call.display_menu
+  end
+
+  def create_rental
+    puts 'Select a book (use the ID please), enter your choice below.'
+    puts
+    @books.each_with_index { |book, index| puts "ID: [#{index}] | Title: #{book.title} - Author #{book.author}" }
+    puts
+    print 'Please enter the number of the required book: '
+    book_selected = gets.chomp.to_i
+
+    puts 'Select a person from the following list (by number, not by ID)'
+    puts
+    @people.each_with_index do |person, index|
+      puts "Number: [#{index}] | Name: #{person.name} - ID: #{person.id} - Age: #{person.age}"
+    end
+    puts
+    print 'Enter the number of the corresponding person: '
+    person_selected = gets.chomp.to_i
+
+    print 'Date: '
+    rental_date = gets.chomp
+    rental = Rental.new(rental_date, @books[book_selected], @people[person_selected])
+    @rentals << rental
+    puts 'Your selected book has been rented'
+    @main_call.display_menu
+  end
+
+  def list_rentals_for_person
+    puts 'Select a person from the following list:'
+    @people.each_with_index do |person, index|
+      puts "Number: [#{index}] | Name: #{person.name} - ID: #{person.id}"
+    end
+
+    person_index = gets.chomp.to_i
+
+    if person_index.negative? || person_index >= @people.length
+      puts 'Invalid person number.'
+      return
+    end
+
+    person = @people[person_index]
+
+    rentals = @rentals.select { |r| r.person == person }
+
+    puts "Rentals for #{person.name}:"
+    rentals.each do |rental|
+      puts "#{rental.book.title} by #{rental.book.author} - #{rental.date}"
+    end
+
+    @main_call.display_menu
+  end
```

Comment:
- [ ] I noticed, when trying to list rentals for a person, that instead of supplying the person's ID, I was required to supply the number or index. Supplying the ID would result in a notice saying "Invalid person number" as you can see in the attached image below 👇 
Please ensure that your app works just like in this [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) by making sure that we can list rentals for a person when we enter the person's ID 😎 

![image](https://github.com/federicaulzurrun/SchoolLibrary/assets/46839548/65b310fe-287e-443a-b0fc-7c2a7e5e08cc)